pkg/cmd/dir.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"k8s.io/klog/v2"
+)
+
+// MkdirRoot creates the root configuration directory for the KCP
+// server. This has to be done early before we start bringing up server
+// components to ensure that we set the initial permissions correctly,
+// since otherwise components will create it as a side-effect.
+func MkdirRoot(dir string) error {
+	if dir == "" {
+		return errors.New("missing root directory configuration")
+	}
+
+	fi, err := os.Stat(dir)
+	if err != nil {
+		klog.Infof("Creating root directory %s", dir)
```

Comment:
logging in a generic helper like this is odd.

Reply:
I found that the config path was a bit non-obvious, so clearly logging the persistent state directory was helpful.